File: server/projects/profile/src/main/java/com/raxim/myscoutee/profile/controller/CalcRestController.java
package com.raxim.myscoutee.profile.controller;

import org.springframework.data.rest.webmvc.RepositoryRestController;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;

import com.google.firebase.messaging.FirebaseMessagingException;
import com.raxim.myscoutee.profile.exception.InvalidScheduleSettingsException;
import com.raxim.myscoutee.profile.repository.mongo.UserRepository;
import com.raxim.myscoutee.profile.service.EventScheduler;

@RepositoryRestController
@RequestMapping("calc")
public class CalcRestController {

    private final EventScheduler eventScheduler;
    private final UserRepository userRepository;

    public CalcRestController(EventScheduler eventScheduler, UserRepository userRepository) {
        this.eventScheduler = eventScheduler;
        this.userRepository = userRepository;
    }

    @GetMapping("/priority_none/{lastTime}")
    public ResponseEntity<Void> calcPriorityNone(@PathVariable String lastTime) {
        try {
            eventScheduler.generateEvents();
        } catch (FirebaseMessagingException | InvalidScheduleSettingsException e) {
            e.printStackTrace();
        }
        return ResponseEntity.ok().build();
    }

    @GetMapping("/messaging/{lastTime}")
    public ResponseEntity<Void> registerMessaging(@PathVariable String lastTime) {
        try {
            eventScheduler.topicRegistration(lastTime);
        } catch (FirebaseMessagingException e) {
            e.printStackTrace();
        }
        return ResponseEntity.ok().build();
    }
}

File: server/projects/profile/src/main/java/com/raxim/myscoutee/profile/controller/CalcRestController.kt
package com.raxim.myscoutee.profile.controller

import com.raxim.myscoutee.profile.repository.mongo.UserRepository
import com.raxim.myscoutee.profile.service.EventScheduler
import org.springframework.data.rest.webmvc.RepositoryRestController
import org.springframework.http.ResponseEntity
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping

@RepositoryRestController
@RequestMapping("calc")
class CalcRestController(private val eventScheduler: EventScheduler, private val userRepository: UserRepository) {

    @GetMapping("/priority_none/{lastTime}")
    fun calcPriorityNone(@PathVariable lastTime: String): ResponseEntity<Void> {

        this.eventScheduler.generateEvents()

        return ResponseEntity.ok().build()
    }

    @GetMapping("/messaging/{lastTime}")
    fun registerMessaging(@PathVariable lastTime: String): ResponseEntity<Void> {
        this.eventScheduler.topicRegistration(lastTime)

        return ResponseEntity.ok().build()
    }
}